code/obj/machinery/turret.dm

```diff
@@ -379,74 +379,55 @@
 /obj/machinery/turretid/attack_ai(mob/user as mob)
 	return attack_hand(user)
 
-/obj/machinery/turretid/attack_hand(mob/user as mob)
-	if (user.getStatusDuration("stunned") || user.getStatusDuration("weakened") || user.stat)
-		return
-
-	if(!in_interact_range(src, user))
-		boutput(user, text("Too far away."))
-		src.remove_dialog(user)
-		user.Browse(null, "window=turretid")
-		return
-
-	src.add_dialog(user)
+/obj/machinery/turretid/ui_interact(mob/user, datum/tgui/ui)
+	if (src.locked)
+		if (!issilicon(usr) && !isAI(usr))
+			boutput(usr, "Control panel is locked!")
+			return
```

Comment:
Would prefer this was handled in the UI rather than on opening. Could lead to some inconsistencies otherwise.